### PR TITLE
Update assumption on lang codes to explicitly reject ISO 639.2

### LINKED ISSUE #1613: Should ACT rules allow 3-letter language codes not in the IANA list
Today I learned, the 3-letter character codes that have a 2-letter counterpart are not in the [IANA registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) which ACT rules use as its list of valid primary language codes. So for example, lang="eng" fails ACT rules. These 3-letter codes are standardised in [ISO 369](https://www.loc.gov/standards/iso639-2/php/code_list.php), however for a reason that I haven't yet figured out they aren't included in the IANA registry.

@goodwitch is currently researching, and I'm hoping we can get a recommendation from her on what to do with this.

### PR DESCRIPTION
**Pull request by Jym77**
Update assumption on `lang` attribute and grandfathered tags to all non-valid tags, and explicitly mention that ISO 639.2 three-letters codes are bad.

Closes issue(s):

- closes #1613 

Need for Final Call:
This will not require a Final Call (changing an assumption)

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [X] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

**Review by carlosapaduarte — APPROVED** [review]

**Review by ajanec01 — APPROVED** [review]

File: _rules/element-lang-matches-default-language-off6ek.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ For each test target, the [primary language][] of its `lang` [attribute value][]
 
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags][] even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags][].
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.2 Language of Parts][sc312]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
Comment (WilcoFiers) [inline _rules/element-lang-matches-default-language-off6ek.md:49]:
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.2 Language of Parts][sc312]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

File: _rules/element-lang-valid-de46e4.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ For each test target, the `lang` [attribute value][] is a [valid language tag][]
 
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags](#valid-language-tag) even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags](#valid-language-tag).
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.2 Language of Parts][sc312]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:46]:
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.2 Language of Parts][sc312]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

File: _rules/html-page-lang-matches-default-ucwvc8.md [hunk 3]
```diff
@@ -50,7 +50,7 @@ For each test target, the [primary language][] of the [valid language tag][] mat
 
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags](#valid-language-tag) even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags](#valid-language-tag).
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
Comment (WilcoFiers) [inline _rules/html-page-lang-matches-default-ucwvc8.md:53]:
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

File: _rules/html-page-lang-valid-bf051a.md [hunk 4]
```diff
@@ -42,7 +42,7 @@ For each test target, the `lang` attribute has a [valid language tag][].
 
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags](#valid-language-tag) even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags](#valid-language-tag).
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
Comment (WilcoFiers) [inline _rules/html-page-lang-valid-bf051a.md:45]:
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

File: _rules/html-page-lang-xml-lang-match-5b7ae0.md [hunk 5]
```diff
@@ -38,7 +38,7 @@ For each test target, the values of the [primary language subtags][], if any exi
 
 - This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags](#valid-language-tag) even if these do not conform to the [BCP 47][] syntax.
 
-- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags](#valid-language-tag).
+- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] or [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```
Comment (WilcoFiers) [inline _rules/html-page-lang-xml-lang-match-5b7ae0.md:41]:
```suggestion
- This rule assumes that only [valid language tags][valid language tag] are enough to satisfy [Success Criterion 3.1.1 Language of Page][sc311]; this notably excludes [grandfathered tags][] and [ISO 639.2][] three-letters codes, both having poor support in assistive technologies.
```

**Review by WilcoFiers — DISMISSED** [review]
Kinda think we should add some failure examples to specifically check that lang="eng" is failed. Probably same for grandfather tags too. WDYT?

**Comment by Jym77** [comment]
> Kinda think we should add some failure examples to specifically check that lang="eng" is failed. Probably same for grandfather tags too. WDYT?

@WilcoFiers @ajanec01 @carlosapaduarte I've added examples with grandfathered and 3-letters codes.

I also noticed that all the links to BCP went to a generic BCP 47 page. I've updated them to RFC 5646. This is not perfect (as BCP numbers are supposed to be more stable than RFC ones), but should be good enough (this one has been here since 2009). Couldn't find a page with the full BCP 47 **which would let me deep link to specific sections** 😕 

---
Adding these examples means the following for tools who actually accept them:
- for the "`lang` is valid" rules, there are some Failed Examples that they may pass. Thus they won't have a correct implementation.
- for the "`lang` match default" rules, there are some Inapplicable Examples that they may pass. This won't really affect the implementation.

File: _rules/element-lang-matches-default-language-off6ek.md [hunk 6]
```diff
@@ -299,17 +299,43 @@ There is no [text inheriting its programmatic language][] from this `p` element
 </html>
 ```
 
+#### Inapplicable Example 6
+
+The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code, hence not a [valid language tag][].
```
Comment (WilcoFiers) [inline _rules/element-lang-matches-default-language-off6ek.md:304]:
```suggestion
The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code. This is not a [valid language tag][].
```

File: _rules/element-lang-matches-default-language-off6ek.md [hunk 7]
```diff
@@ -299,17 +299,43 @@ There is no [text inheriting its programmatic language][] from this `p` element
...
+<html lang="en">
+	<body>
+		<p lang="eng">I love ACT rules!</p>
+	</body>
+</html>
+```
+
+#### Inapplicable Example 7
+
+The `lang` [attribute value][] of this `p` element is a [grandfathered tag][grandfathered tags], hence not a [valid language tag][].
```
Comment (WilcoFiers) [inline _rules/element-lang-matches-default-language-off6ek.md:316]:
```suggestion
The `lang` [attribute value][] of this `p` element is a [grandfathered tag][grandfathered tags]. This is not a [valid language tag][].
```

File: _rules/element-lang-matches-default-language-off6ek.md [hunk 8]
```diff
@@ -299,17 +299,43 @@ There is no [text inheriting its programmatic language][] from this `p` element
...
+
+```html
+<html lang="lb">
+	<body>
+		<p lang="i-lux">Lëtzebuerg ass e Land an Europa.</p>
```
Comment (WilcoFiers) [inline _rules/element-lang-matches-default-language-off6ek.md:321]: Owww, good one! I like that better than having Klingon in our examples!

File: _rules/element-lang-valid-de46e4.md [hunk 9]
```diff
@@ -238,6 +238,30 @@ This `div` element has an invalid `lang` [attribute value][]. The [accessible na
 </html>
 ```
 
+#### Failed Example 8
+
+The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code, hence not a [valid language tag][].
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:243]: How about making this a little more explicit:

```suggestion
The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code. This is not part of the [language subtag registry][].
```

File: _rules/element-lang-valid-de46e4.md [hunk 10]
```diff
@@ -238,6 +238,30 @@ This `div` element has an invalid `lang` [attribute value][]. The [accessible na
...
+<html lang="en">
+	<body>
+		<p lang="eng">I love ACT rules!</p>
+	</body>
+</html>
+```
+
+#### Failed Example 9
+
+The `lang` [attribute value][] of this `p` element is a [grandfathered tag][grandfathered tags], hence not a [valid language tag][].
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:255]:
```suggestion
The `lang` [attribute value][] of this `p` element is a [grandfathered tag][grandfathered tags]. This is not part of the [language subtag registry][].
```

File: _rules/html-page-lang-valid-bf051a.md [hunk 11]
```diff
@@ -97,6 +97,30 @@ This `html` element has a `lang` attribute whose value is not a [valid language
...
 ```
 
+#### Failed Example 3
+
+The `lang` attribute of this page is an [iso 639.2][] three letters code, hence not a [valid language tag][].
+
+```html
+<html lang="eng">
+	<body>
+		<p lang="en">I love ACT rules!</p>
```
Comment (WilcoFiers) [inline _rules/html-page-lang-valid-bf051a.md:107]: What's the reason for putting in the `lang` on this `p` element? I think this is clearer as an example if it isn't there. One could even make the case that in this particular example it isn't actually an error. Not that I want to have that discussion.

```suggestion
		<p>I love ACT rules!</p>
```

File: _rules/html-page-lang-valid-bf051a.md [hunk 12]
```diff
@@ -97,6 +97,30 @@ This `html` element has a `lang` attribute whose value is not a [valid language
...
+```
+
+#### Failed Example 4
+
+The `lang` attribute of this page is a [grandfathered tag][grandfathered tags], hence not a [valid language tag][].
+
+```html
+<html lang="i-lux">
+	<body>
+		<p lang="lb">Lëtzebuerg ass e Land an Europa.</p>
```
Comment (WilcoFiers) [inline _rules/html-page-lang-valid-bf051a.md:119]:
```suggestion
		<p>Lëtzebuerg ass e Land an Europa.</p>
```

**Review by WilcoFiers — CHANGES_REQUESTED** [review]
Great work! Just a couple of editorial suggestions.

File: _rules/element-lang-valid-de46e4.md (hunk 9 above)
Comment (Jym77) [reply _rules/element-lang-valid-de46e4.md:243]: I think we should stick with the "not a valid tag" explanation as we do for grandfathered tags (https://github.com/act-rules/act-rules.github.io/pull/1650/files#r673783251) (for the sake of consistency). WDYT?
```suggestion
The `lang` [attribute value][] of this `p` element is an [iso 639.2][] three letters code. This is not a [valid language tag][].
```

File: _rules/html-page-lang-valid-bf051a.md (hunk 11 above)
Comment (Jym77) [reply _rules/html-page-lang-valid-bf051a.md:107]: Without the `lang` attribute on the paragraph, the page is also failing 3.1.2 (on top of 3.1.1 that the rule test).

**Review by WilcoFiers — APPROVED** [review]

**Review by daniel-montalvo — APPROVED** [review]

**Pull request merged**